src/Compilers/Core/Portable/StrongName/PortableStrongNameProvider.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Cci;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal class PortableStrongNameProvider : StrongNameProvider
+    {
+        internal readonly ImmutableArray<string> _keySearchPaths;
+        internal IOOperations IOOp { get; private set; }
+
+        public override bool Equals(object other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (other.GetType() != this.GetType()) {
+                return false;
```

Comment:
>   return false; [](start = 13, length = 16)

This can all be simplified to 

``` c#
var o = other as PortableStrongNameProvider;
if (o == null) 
{
  return false;
}
...
```

Reply:
?

---
In reply to: [137319197](https://github.com/dotnet/roslyn/pull/21718#discussion_r137319197) [](ancestors = 137319197)

Reply:
```
            if (!(obj is PortableStrongNameProvider other))
            {
                return false;
            }

            return FileSystem == other.FileSystem &&
                _keyFileSearchPaths.SequenceEqual(other._keyFileSearchPaths);
```